collector-bitmovin-player/src/main/java/com/bitmovin/analytics/bitmovin/player/BitmovinSdkAdAdapter.kt

```diff
@@ -14,25 +14,29 @@ import com.bitmovin.player.api.event.listener.*
  */
 class BitmovinSdkAdAdapter(val bitmovinPlayer: BitmovinPlayer, val adAnalytics: BitmovinAdAnalytics) : AdAdapter {
 
-    private val adFactory: AdFactory = AdFactory()
-    private val adBreakFactory: AdBreakFactory = AdBreakFactory()
+    private val adMapper: AdMapper = AdMapper()
+    private val adBreakMapper: AdBreakMapper = AdBreakMapper()
     private val adQuartileFactory: AdQuartileFactory = AdQuartileFactory()
 
 
     private val onAdStartedListener = OnAdStartedListener {
-        if(it.ad != null){
-            adAnalytics.onAdStarted(adFactory.FromPlayerAd(it.ad!!))
+        if(it.ad == null){
+            return@OnAdStartedListener
         }
+
+        adAnalytics.onAdStarted(adMapper.FromPlayerAd(it.ad!!))
     }
 
     private val onAdFinishedListener = OnAdFinishedListener {
         adAnalytics.onAdFinished()
     }
 
     private val onAdBreakStartedListener = OnAdBreakStartedListener {
-        if (it.adBreak != null){
-            adAnalytics.onAdBreakStarted(adBreakFactory.FromPlayerImaAdBreak(it.adBreak!! as com.bitmovin.player.model.advertising.ima.ImaAdBreak))
+        if (it.adBreak == null){
+            return@OnAdBreakStartedListener
         }
+
+        adAnalytics.onAdBreakStarted(adBreakMapper.FromPlayerAdBreak(it.adBreak!!))
```

Comment:
Elvis operator would remove the unsafe `!!`

Reply:
Agree, comes with the next commit